### PR TITLE
Adding 600-series driver

### PR DESCRIPTION
**Pull request by pazaan**
Add support for 600-series driver.

### CODE REVIEW

File: app/reducers/devices.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+
 const devices = {
   carelink: {
     instructions: ['Import from CareLink', '(We will not store your credentials)'],
     isFetching: false,
     key: 'carelink',
     name: 'Medtronic',
     // for the device selection list
-    selectName: 'Medtronic (from CareLink)',
+    selectName: 'Medtronic (CareLink import)',
```
Comment (pazaan) [inline app/reducers/devices.js]: @gniezen FYI, https://trello.com/c/BF1URAvH covers the copy changes in this file that are not related to 600-series.

Comment (gniezen) [reply app/reducers/devices.js]: As CareLink is hidden now, this specific change doesn't matter. Removing the `-` for 5/7-series LGTM.

File: lib/drivers/medtronic600/medtronic600Driver.js [hunk 2]
```diff
@@ -0,0 +1,2537 @@
...
+ * == BSD2 LICENSE ==
+ */
+
+// I *like* for..in
+/* eslint no-restricted-syntax: [0, "ForInStatement"] */
+const _ = require('lodash');
+const sundial = require('sundial');
+const crypto = require('crypto');
+const iconv = require('iconv-lite');
+const promisify = require('util-promisify'); // TODO: replace with built-in once we move to Node v8
```
Comment (gniezen) [inline lib/drivers/medtronic600/medtronic600Driver.js]: We are now on Node v8?

File: test/lib/medtronic600/testMedtronic600Simulator.js [hunk 3]
```diff
@@ -0,0 +1,1498 @@
...
+        simulator.basal(basal1);
+        simulator.basal(basal2);
+        simulator.finalBasal();
+
+        expect(simulator.getEvents()).deep.equals([expectedFirstBasal.done(),
+          expectedSecondBasal.done(), expectedThirdBasal.done(),
+        ]);
+      });
+
+      /* TODO - add test for Caty's 11th January 2017 data.
```
Comment (gniezen) [inline test/lib/medtronic600/testMedtronic600Simulator.js]: Do you want to remove this?
